src/Provider/URLBuilder.js

```diff
@@ -63,9 +63,19 @@ export default {
      * @return {string} the formed url
      */
     xyz: function xyz(coords, source) {
-        return subDomains(source.url.replace(/(\$\{z\}|%TILEMATRIX)/, source.tileMatrixCallback(coords.zoom))
-            .replace(/(\$\{y\}|%ROW)/, coords.row)
-            .replace(/(\$\{x\}|%COL)/, coords.col));
+        if (typeof source.url === 'string') {
+            return subDomains(source.url.replace(/(\$\{z\}|%TILEMATRIX)/, source.tileMatrixCallback(coords.zoom))
+                .replace(/(\$\{y\}|%ROW)/, coords.row)
+                .replace(/(\$\{x\}|%COL)/, coords.col));
+        } else if (Array.isArray(source.url)) {
```

Comment:
instead of dealing with the multi url inside this function it should be done earlier and, this function should be called x times (x being the length of the array)